flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java

```diff
@@ -160,25 +168,17 @@ public void testAsyncErrorRethrownOnInvoke() throws Throwable {
                 .get(0)
                 .onCompletion(null, new Exception("artificial async exception"));
 
-        try {
-            testHarness.processElement(new StreamRecord<>("msg-2"));
-        } catch (Exception e) {
-            // the next invoke should rethrow the async exception
-            assertThat(e.getCause().getMessage()).contains("artificial async exception");
-
-            // test succeeded
-            return;
-        }
-
-        fail("unknown failure");
+        // the next invoke should rethrow the async exception
+        assertThatThrownBy(() -> testHarness.processElement(new StreamRecord<>("msg-2")))
+                .hasStackTraceContaining("artificial async exception");
```

Comment:
Here it is not one-to-one migration since it's started to check the, whole trace instead of just a cause